src/hera.cpp

```diff
@@ -425,7 +505,8 @@ evmc_set_option_result hera_set_option(
   if (strcmp(name, "engine") == 0) {
     auto it = wasm_engine_map.find(value);
     if (it != wasm_engine_map.end()) {
-      hera->engine = it->second();
+      wasmEngineCreateFn = it->second;
+      hera->engine = wasmEngineCreateFn();
```

Comment:
Sorry, you were correct, `.execute()` is stateless at the moment. Would you mind if I remove the changes regarding the engines and merge this PR?

And then could you create a new PR changing the handling of the engine, because that might be a bug currently, but want to have @chfast's opinion.

Reply:
Sure, feel free to modify the PR.